cmd/operator-controller/main.go

```diff
@@ -408,9 +414,11 @@ func run() error {
 	}
 
 	helmApplier := &applier.Helm{
-		ActionClientGetter:  acg,
-		Preflights:          preflights,
-		BundleToHelmChartFn: convert.RegistryV1ToHelmChart,
+		ActionClientGetter:         acg,
+		Preflights:                 preflights,
+		BundleToHelmChartFn:        convert.RegistryV1ToHelmChart,
+		PreAuthorizer:              authorization.NewRBACPreAuthorizer(mgr.GetClient()),
+		EnablePreflightPermissions: features.OperatorControllerFeatureGate.Enabled(features.PreflightPermissions),
```

Comment:
I may have suggested this structure in a comment, but seeing it now, we could simplify the struct by dropping `EnablePreflightPermissions` and then just conditionally setting `PreAuthorizer` only if the feature gate is enabled.  And then the helm applier could conditionally use the `PreAuthorizer`, only if it is non-nil.